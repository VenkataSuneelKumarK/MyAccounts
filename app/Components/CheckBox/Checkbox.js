/**
 * Created by kanamars on 27/10/19.
 */
import React from 'react';

const checkBox = props =>{
    return (
        <div>
            <input  id={props.value.id} type="checkbox" checked={props.value.isChecked} />
            <label htmlFor={props.value.id}>{props.value.label}</label>
        </div>
    )
};

export default checkBox;

// onChange={props.handleChange}