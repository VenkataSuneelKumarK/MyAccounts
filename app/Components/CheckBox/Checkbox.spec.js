/**
 * Created by kanamars on 29/10/19.
 */
import React from 'react';

import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import CheckBox from './Checkbox';
configure({adapter: new Adapter()});

describe('<CheckBox/>', function () {
    let wrapper, value, onchangeHandler;

    beforeEach(() => {
        value = {
            id : "1",
            label: "Saving Account",
            isChecked: false
        };
        onchangeHandler = () => {};
        wrapper = shallow(<CheckBox value={value} key="1" onchange={onchangeHandler} filterKey="Account"/>);
    });
    it('should render <CheckBox /> element', () => {
        // console.log("wrapper", wrapper.debug());
        expect(wrapper.find("input")).toHaveLength(1);
        expect(wrapper.find("label")).toHaveLength(1);
        expect(wrapper.find("label[htmlFor='Account&1']")).toHaveLength(1);
    });
    it("should snapshot", function () {
        const tree = renderer
            .create(<CheckBox value={value} key="1" onchange={onchangeHandler} filterKey="Account"/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});