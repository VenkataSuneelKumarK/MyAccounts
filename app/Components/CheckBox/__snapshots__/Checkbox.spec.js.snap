// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckBox/> should snapshot 1`] = `
<div>
  <input
    checked={false}
    id="Account&1"
    onChange={[Function]}
    type="checkbox"
  />
  <label
    htmlFor="Account&1"
  >
    Saving Account
  </label>
</div>
`;
